src/components/NavBar/LanguageSelector/index.tsx

```diff
@@ -1,11 +1,21 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import Moment from 'moment';
+import { extendMoment } from 'moment-range';
 import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core';
-import { languages } from '../../../i18n';
+import { isLocalLanguageChosen, languages, safeTranslate } from '../../../i18n';
 
 function LanguageSelector({ classes }: LanguageSelectorProps) {
-  const { i18n } = useTranslation();
+  const { t, i18n } = useTranslation();
 
+  const moment = extendMoment(Moment as any);
+
+  const handleChangeLanguage = (lng: string): void => {
+    i18n.changeLanguage(lng);
+    moment.locale(
+      isLocalLanguageChosen(i18n) ? safeTranslate(t, 'date_locale') : 'en',
```

Comment:
You can probably just load `safeTranslate(t, 'date_locale')` since the default will be 'en'

Reply:
not sure why but for english language, `safeTranslate(t, 'date_locale')` return 'date_locale' instead of 'en', that's why i did this